src/test/java/uk/gov/pay/directdebit/payments/dao/PaymentViewDaoIT.java

```diff
@@ -214,9 +223,33 @@ public void shouldReturnOnePaymentViewWhenPaymentCreated() {
                     .withAmount(1000L + i)
                     .insert(testContext.getJdbi());
         }
-        PaymentViewSearchParams searchParams = new PaymentViewSearchParams(gatewayAccountFixture.getExternalId(),
-                2L, 100L, null, null, null, null, null, null, searchDateParams);
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams(gatewayAccountFixture.getExternalId())
+                .withPage(2L)
+                .withDisplaySize(100L)
+                .withSearchDateParams(searchDateParams);
         List<PaymentView> viewList = paymentViewDao.searchPaymentView(searchParams);
         assertThat(viewList.size(), is(1));
     }
+    
+    @Test
+    public void shouldReturnCount_whenRecordsExist() {
+        GatewayAccountFixture gatewayAccountFixture = GatewayAccountFixture.aGatewayAccountFixture().insert(testContext.getJdbi());
+        MandateFixture mandateFixture = MandateFixture.aMandateFixture().withGatewayAccountFixture(gatewayAccountFixture).insert(testContext.getJdbi());
+        aPayerFixture()
+                .withMandateId(mandateFixture.getId())
+                .insert(testContext.getJdbi());
+        for (int i = 0; i < 3; i++) {
+            aTransactionFixture()
+                    .withId((long) i)
+                    .withMandateFixture(mandateFixture)
+                    .withReference("important reference " + i)
+                    .withDescription("description " + i)
+                    .withAmount(1000L + i)
+                    .insert(testContext.getJdbi());
+        }
+        PaymentViewSearchParams searchParams = new PaymentViewSearchParams(gatewayAccountFixture.getExternalId())
+                .withMandateId(mandateFixture.getExternalId());
+        Long count = paymentViewDao.getPaymentViewCount(searchParams);
+        assertThat(count, is(3L));
+    }
 }
```

Comment:
think you need one more test in here just to prove that searching by mandate id works - ie have two payments with different mandate ids and search for payment by mandate id and check you get the right one back